src/components/approval-type/ApprovalType.vue

```diff
@@ -0,0 +1,169 @@
+<template>
+  <div id="approval-type">
+    <v-row no-gutters>
+      <v-col cols="12" sm="3" class="pr-4">
+        <label id="approval-type-label">Approval Type</label>
+      </v-col>
+      <v-col cols="12" sm="9">
+        <v-radio-group class="payment-group" v-model="approvalTypeSelected" @change="radioButtonChanged">
+          <!-- COURT ORDER radio button -->
+          <v-radio id="court-order-radio" class="mb-0"
+             :label="getRadioText(ApprovalTypes.VIA_COURT_ORDER)"
+             :value="ApprovalTypes.VIA_COURT_ORDER"
+          />
+          <v-form ref="courtNumRef" id="court-num-form" v-model="valid" class="mt-4 ml-8">
+            <v-text-field
+              @input="courtOrderNumberChanged"
+              id="court-order-number-input"
+              v-model="courtOrderNumber"
+              label="Court Order Number"
+              :rules="courtOrderNumRules"
+              :disabled="approvalTypeSelected === ApprovalTypes.VIA_REGISTRAR"
+              @update:error="emitValidationError"
+              filled
+            />
+          </v-form>
+          <!-- REGISTRAR radio button -->
+          <v-radio id="registrar-radio" class="mb-0 pt-2"
+             :label="getRadioText(ApprovalTypes.VIA_REGISTRAR)"
+             :value="ApprovalTypes.VIA_REGISTRAR"
+             v-if="!isCourtOrderOnly"
+          />
+        </v-radio-group>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { Component, Emit, Prop, Watch } from 'vue-property-decorator'
+import { FormIF } from '@bcrs-shared-components/interfaces'
+import { types } from "sass";
+import Null = types.Null;
+
+@Component({})
+export default class ApprovalType extends Vue {
+  // Refs
+  $refs!: Vue['$refs'] & {
+    courtNumRef: FormIF
+  }
+
+  ApprovalTypes = {
+    VIA_COURT_ORDER: 'VIA COURT ORDER',
+    VIA_REGISTRAR: 'VIA REGISTRAR'
+  }
+
+  /** Draft court order number. */
+  @Prop({ default: '' }) readonly draftCourtOrderNumber!: string
+
+  /** Whether approved by the registrar. */
+  @Prop({ default: false }) readonly approvedByRegistrar!: boolean
+
+  /** text used in radio options . */
+  @Prop({ default: false }) readonly isConversionToFullRestoration!: boolean
```

Comment:
I counsel against identifying a specific filing type here... it would be better to use a flag named after the desired behaviour. (This component should not know what's going on in its parent.)

Also, comment is outdated.

Reply:
Good call, I'll fix